text/068-coqide-split.md

```diff
@@ -0,0 +1,102 @@
+
+- what features / changes to include,
+- what technology / internal changes to make,
+- which versions of Coq to support,
+- how to review and merge changes (in particular, we expect that the reviewing process will become more lightweight than it is currently in the Coq repository, with the ability, in particular, to self-merge some pull requests),
+- when and how frequently to release new CoqIDE versions (keeping in mind that some releases will be mandated by the Coq Platform inclusion).
+
+A recommendation to CoqIDE maintainers is to quickly make CoqIDE compatible with multiple Coq versions (at least when the XML protocol doesn't change). This technically already works with the limitation that CoqIDE doesn't check for the version of Coq: https://github.com/coq/coq/issues/16521.
+
+### Long-term plans
+
+As explained above, members of the Coq team are currently developing support for LSP, along with VS Code extensions. There are two projects in this direction: 
+- [VsCoq2](https://github.com/coq-community/vscoq), a project for a stable Coq IDE led by Inria engineers and supported by stable Inria funding, and 
+- [coq-lsp](https://github.com/ejgallego/coq-lsp), a research project led by an Inria researcher and supported by research grants. 
+
+The two projects are based on different architectural choices and allow exploring different directions, but in both cases, they work with no dependency on the STM (VsCoq2 is based on a [new document manager](https://github.com/Coq-community/vscoq/blob/main/docs/developers.md#architecture) that can be considered as an evolved, more modular, and smaller STM, while coq-lsp is based on an incremental document checking engine called Flèche). There are hopes that despite the different architecture and implementation, the specification of the LSP extension that they implement can eventually be shared. coq-lsp is already available for use with recent versions of Coq, and VsCoq2 is planned to become available with Coq 8.18.
+
+When Coq LSP support becomes stable enough, maintainers of IDE support packages for Coq (e.g., Proof General for Emacs, Coqtail for Vim) will be encouraged to migrate to using this protocol. We expect that Coqtail will be one of the first IDEs to migrate, as it currently depends on coqidetop and the XML protocol (and such a migration has already [seen experiments](https://github.com/whonore/Coqtail/pull/323)).
+
+When no major IDE besides CoqIDE relies on the XML protocol anymore, the Coq team will start considering removal of coqidetop from the Coq repository. The decision to remove coqidetop will be made after first evaluating whether all the important features that the Coq team wishes to continue supporting have been made available in other IDEs or through other IDE-related protocols. At this point, it will then become time for CoqIDE maintainers to decide what to do regarding the long-term future of CoqIDE. Several options will be available:
+
```

Comment:
That strategy seems too much in a lock-step to me, and thus of high risk. IMHO instead of a sudden "removal", a more gradual strategy should be used, in particular by gradually deprecating the most problematic features which can be done without breaking 3rd party systems too much.

Reply:
Removal of coqidetop is not as much of a breaking change as the removal of the STM APIs it relies on (coqidetop could be simply moved to the CoqIDE split repo). The feature removal you are talking about concerns mostly the underlying API, and we tried to convey this idea that features could be removed gradually instead of the whole component in the next paragraph on the STM.

Reply:
I saw that yes. Maybe what I mean here is that for example, it is easy to reimplement most of Stm API on top of Flèche, and there are other API bits that I'd personally remove.

So in a sense it is a viable alternative to keep a STM-like API on top of Flèche, in fact I'm willing to do this depending on how the SerAPI migration goes, tho after having a look at what users need I'm convienced they want yet another API for their high-throughtpout use cases.

Reply:
Do you have a pointer to which features are the problematic ones? TBH I don't have a clear idea of what features the stm has.

Reply:
I have some notes, maybe that would be a good topic for the call (too late for this week tho), or even a CEP ?

Reply:
Whatever format you prefer.